src/components/Login/CreateUserForm.js

```diff
@@ -166,71 +158,64 @@ export default class CreateUserForm extends React.Component {
     return addBreak ? <br /> : null;
   };
 
-  updateUsername = (username) => this.setState({ username });
+  const updateUsername = (newusername) => setusername(newusername);
 
-  updatePassword = (password) => this.setState({ password });
+  const updatePassword = (newpassword) => setpassword(newpassword);
 
-  updateConfirmPassword = (confirmPassword) => this.setState({ confirmPassword });
+  const updateConfirmPassword = (newconfirmPassword) => setconfirmPassword(newconfirmPassword);
 
-  renderInputs = () => (
+  const renderInputs = () => (
     <div className="login-form-input-list">
       <div>
-        <LoginInput
-          type="Username"
-          data={this.state.username}
-          waiting={this.state.waiting}
-          onChange={this.updateUsername}
-        />
-        <LoginInput
-          type="Password"
-          data={this.state.password}
-          waiting={this.state.waiting}
-          onChange={this.updatePassword}
-        />
+        <LoginInput type="Username" data={username} waiting={waiting} onChange={updateUsername} />
+        <LoginInput type="Password" data={password} waiting={waiting} onChange={updatePassword} />
         <LoginInput
           type="Confirm Password"
-          data={this.state.confirmPassword}
-          waiting={this.state.waiting}
-          onChange={this.updateConfirmPassword}
+          data={confirmPassword}
+          waiting={waiting}
+          onChange={updateConfirmPassword}
         />
-        {this.renderErrorMessage(this.state.usernameMessage)}
-        {this.renderErrorMessage(this.state.passwordMessage)}
-        {this.renderErrorMessage(this.state.errorMessage, true)}
+        {renderErrorMessage(usernameMessage)}
+        {renderErrorMessage(passwordMessage)}
+        {renderErrorMessage(errorMessage, true)}
       </div>
     </div>
   );
 
-  renderAction = () => {
-    if (this.state.waiting) {
+  const renderAction = () => {
+    const { themeColor, textColor } = props;
+    if (waiting) {
       return (
         <div className="login-form-loader">
-          <RingLoader color={this.props.themeColor} size={80} loading />
+          <RingLoader color={themeColor} size={80} loading />
         </div>
       );
     }
     const unclickedStyle = {
       backgroundColor: 'white',
-      borderColor: this.props.themeColor,
+      borderColor: themeColor,
       borderWidth: 'medium',
       borderRadius: '4px',
       color: 'black',
     };
 
     const clickedStyle = {
-      backgroundColor: this.props.themeColor,
-      borderColor: this.props.themeColor,
+      backgroundColor: themeColor,
+      borderColor: themeColor,
       borderWidth: 'medium',
       borderRadius: '4px',
-      color: this.props.textColor,
+      color: textColor,
     };
+
+    const [hoverButton, sethoverButton] = useState();
```

Comment:
This should be in the main component body, for consistency and so that all the state variables are easily visible.